jdaviz/configs/cubeviz/helper.py

```diff
@@ -34,21 +34,35 @@ def _set_spectrum_x_axis(self, msg):
                         viewer.state.reset_limits()
                     break
             else:
-                viewer.state.x_att_pixel = ref_data.id["Pixel Axis 2 [x]"]
+                viewer.state.x_att = ref_data.id["Pixel Axis 2 [x]"]
+                viewer.state.reset_limits()
 
-    def load_data(self, data, data_label=None, **kwargs):
+    def load_data(self, data, data_label=None, override_cube_limit=False, **kwargs):
         """
         Load and parse a data cube with Cubeviz.
         (Note that only one cube may be loaded per Cubeviz instance.)
 
         Parameters
         ----------
-        data : str or `~astropy.io.fits.HDUList`
-            A string file path or astropy FITS object pointing to the
-            data cube.
+        data : str, `~astropy.io.fits.HDUList`, or ndarray
+            A string file path, astropy FITS object pointing to the
+            data cube, or a Numpy array cube.
+
+        data_label : str or `None`
+            Data label to go with the given data. If not given,
+            one will be automatically generated.
+
+        override_cube_limit : bool
+            Override internal cube count limitation and load the data anyway.
+            Setting this to `True` is not recommended unless you know what
+            you are doing.
+
+        **kwargs : dict
+            Extra keywords accepted by Jdaviz application-level parser.
+
         """
-        if len(self.app.state.data_items) != 0:
-            raise RuntimeError('only one cube may be loaded per Cubeviz instance')
+        if not override_cube_limit and len(self.app.state.data_items) != 0:
```

Comment:
I know I should not be giving out hammer to the toes but I need this so I can load two arbitrary arrays into different viewers. Not advertising it in change log though.